src/lightning/pytorch/strategies/xla_fsdp.py

```diff
@@ -0,0 +1,573 @@
+        **kwargs: Any,
+    ) -> None:
+        if not _XLA_AVAILABLE:
+            raise ModuleNotFoundError(str(_XLA_AVAILABLE))
+        super().__init__(
+            accelerator=accelerator,
+            parallel_devices=parallel_devices,
+            cluster_environment=XLAEnvironment(),
+            checkpoint_io=checkpoint_io,
+            precision_plugin=precision_plugin,
+        )
+        self._checkpoint_io: Optional[CheckpointIO]
+        self._backward_sync_control = _XLAFSDPBackwardSyncControl()
+
+        self._auto_wrap_policy = auto_wrap_policy
+        self._activation_checkpointing_policy = activation_checkpointing_policy
+        self._fsdp_kwargs = kwargs
+        self._state_dict_type = state_dict_type
+        self._sequential_save = sequential_save
+        self._launched = False
+
+    @property
+    def checkpoint_io(self) -> CheckpointIO:
+        if self._checkpoint_io is None:
+            self._checkpoint_io = XLACheckpointIO()
+        elif isinstance(self._checkpoint_io, _WrappingCheckpointIO):
+            self._checkpoint_io.checkpoint_io = XLACheckpointIO()
+
+        return self._checkpoint_io
+
+    @checkpoint_io.setter
+    def checkpoint_io(self, io: CheckpointIO) -> None:
+        self._checkpoint_io = io
+
+    @property
+    def root_device(self) -> torch.device:
+        if not self._launched:
+            raise RuntimeError("Accessing the XLA device before processes have spawned is not allowed.")
+        import torch_xla.core.xla_model as xm
+
+        return xm.xla_device()
+
+    @property
+    def num_processes(self) -> int:
+        return len(self.parallel_devices) if self.parallel_devices is not None else 0
+
+    @property
+    def global_rank(self) -> int:
+        return super().global_rank if self._launched else 0
+
+    @property
+    def local_rank(self) -> int:
+        return super().local_rank if self._launched else 0
+
+    @property
+    def node_rank(self) -> int:
+        return super().node_rank if self._launched else 0
+
+    @property
+    def world_size(self) -> int:
+        return super().world_size if self._launched else 1
+
+    def _configure_launcher(self) -> None:
+        self._launcher = _XLALauncher(self)
+
+    def setup_environment(self) -> None:
+        assert self.parallel_devices is not None
+        if _using_pjrt() and len(self.parallel_devices) == 1:
+            # spawning only 1 device with PjRT is not supported:
+            # https://github.com/Lightning-AI/lightning/pull/17408#discussion_r1170671732
+            raise NotImplementedError(
+                f"The {type(self).__name__} does not support running on a single device with the PjRT runtime."
+                " Try using all devices or the `SingleDeviceXLAStrategy` strategy"
+            )
+
+        self._launched = True
+        rank_zero_only.rank = self.global_rank
+        super().setup_environment()
+
+    def setup(self, trainer: "pl.Trainer") -> None:
```

Comment:
By comparing this with https://github.com/Lightning-AI/lightning/blob/master/src/lightning/pytorch/strategies/xla.py#L113-L137, it looks like we are missing the precision plugin setup.

Also, we should check if we need to do something about tied parameters

Reply:
I added `self.setup_precision_plugin()`